contracts/RolesRegistry/SftRolesRegistry.sol

```diff
@@ -0,0 +1,245 @@
+// SPDX-License-Identifier: CC0-1.0
+
+pragma solidity 0.8.9;
+
+import { IERCXXXX } from './interfaces/IERCXXXX.sol';
+import { IERC165 } from '@openzeppelin/contracts/utils/introspection/IERC165.sol';
+import { IERC1155 } from '@openzeppelin/contracts/token/ERC1155/IERC1155.sol';
+import { IERC1155Receiver } from '@openzeppelin/contracts/token/ERC1155/IERC1155Receiver.sol';
+import { ERC1155Holder, ERC1155Receiver } from '@openzeppelin/contracts/token/ERC1155/utils/ERC1155Holder.sol';
+import { ERC165Checker } from '@openzeppelin/contracts/utils/introspection/ERC165Checker.sol';
+
+// Semi-fungible token (SFT) registry with only one role (UNIQUE_ROLE)
+contract SftRolesRegistry is IERCXXXX, ERC1155Holder {
+    bytes32 public constant UNIQUE_ROLE = keccak256('UNIQUE_ROLE');
+
+    // grantor => tokenAddress => operator => isApproved
+    mapping(address => mapping(address => mapping(address => bool))) public tokenApprovals;
+
+    // nonce => DepositInfo
+    mapping(uint256 => DepositInfo) public deposits;
+
+    // nonce => role => RoleAssignment
+    mapping(uint256 => mapping(bytes32 => RoleData)) internal roleAssignments;
```

Comment:
Since it's using only one roles, it will always be nonce => UNIQUE_ROLE => RoleAssignment.

So, i think it do not need `bytes32` in mapping

```suggestion
    mapping(uint256 => RoleData) internal roleAssignments;
```